### DISCUSSION TITLE
App freezes randomly

### DISCUSSION THREAD

**Discussion by Soif2Sang** [flet-dev/flet#2175] [Q&A]
**Description**

Hello there, Few weeks ago I created an issue about this but I was unable to isolate the issue, I created a repository containing just enough code to create my use case.
Very easy, there's few rows, clicks on the "start" buttons that will start fake jobs that only aims to fill the logs and create timers.
Click on the gear, wait few minutes and see that the apps suddenly stops refreshing and NO exception are raised. 
Note: if you don't click on the gear, the application will never freeze.

**Code example to reproduce the issue:**

https://github.com/Soif2Sang/flet_bug

**Describe the results you received:**

The application stops refreshing, page.update() does not work anymore

**Describe the results you expected:**

page.update() to keep working

**Flet version (`pip show flet`)**:

This bug occurs for all flet above 0.7 (I did not test bellow)

**Operating system**:

Windows & mac suffers from this

**Comment by Soif2Sang** [comment]
Note to anyone that can solve this problem, I can pay.

**Comment by FeodorFitsner** [comment]
Interesting!

**Comment by FeodorFitsner** [comment]
The app in that repo doesn't look runnable - I see some class definitions (like FileSingleton for example) are missing. Could you take a look please?

**Comment by Soif2Sang** [comment]
It is runnable, these files are just from an other repo (only few functions are needed but I did not filter it).
I'm pretty sure you only need flet & decohints dependencies

**Comment by Soif2Sang** [comment]
@FeodorFitsner any news about this?
